failed to build on macos 10.12

```
Some errors occurred: E0432, E0658.
For more information about an error, try `rustc --explain E0432`.
error: Could not compile `futures-core`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `bandwhich v0.6.0`, intermediate artifacts can be found at `/var/folders/37/jbt828fj49932z8nspl09nc00000gn/T/cargo-installz0gwr1`

```

full output:
```
   Compiling futures-core v0.3.1
error[E0432]: unresolved import `core::task::Context`
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:5:18
  |
5 | use core::task::{Context, Poll};
  |                  ^^^^^^^ no `Context` in `task`

error[E0432]: unresolved import `core::task::Context`
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:5:18
  |
5 | use core::task::{Context, Poll};
  |                  ^^^^^^^ no `Context` in `task`

error[E0432]: unresolved imports `core::task::Context`, `core::task::RawWaker`, `core::task::RawWakerVTable`
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/mod.rs:9:22
  |
9 | pub use core::task::{Context, Poll, Waker, RawWaker, RawWakerVTable};
  |                      ^^^^^^^               ^^^^^^^^  ^^^^^^^^^^^^^^ no `RawWakerVTable` in `task`
  |                      |                     |
  |                      |                     no `RawWaker` in `task`
  |                      no `Context` in `task`

error[E0658]: use of unstable library feature 'alloc': this library is unlikely to be stabilized in its current form or name (see issue #27783)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/lib.rs:20:1
   |
20 | extern crate alloc;
   | ^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/lib.rs:23:39
   |
23 | #[doc(hidden)] pub use self::future::{Future, FusedFuture, TryFuture};
   |                                       ^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:5:27
  |
5 | use core::task::{Context, Poll};
  |                           ^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:7:9
  |
7 | pub use core::future::Future;
  |         ^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:12:55
   |
12 | pub type BoxFuture<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + Send + 'a>>;
   |                                                       ^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:16:60
   |
16 | pub type LocalBoxFuture<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + 'a>>;
   |                                                            ^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:26:24
   |
26 | pub trait FusedFuture: Future {
   |                        ^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:57:22
   |
57 | pub trait TryFuture: Future + private_try_future::Sealed {
   |                      ^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:76:23
   |
76 |     where F: ?Sized + Future<Output = Result<T, E>>
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:72:10
   |
72 |     ) -> Poll<Result<Self::Ok, Self::Error>>;
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:82:64
   |
82 |     fn try_poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<F::Output> {
   |                                                                ^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:48:9
   |
48 |     use super::Future;
   |         ^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:52:50
   |
52 |     impl<F, T, E> Sealed for F where F: ?Sized + Future<Output = Result<T, E>> {}
   |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:5:27
  |
5 | use core::task::{Context, Poll};
  |                           ^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:61:10
   |
61 |     ) -> Poll<Option<Self::Item>>;
   |          ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:181:12
    |
181 |         -> Poll<Option<Result<Self::Ok, Self::Error>>>;
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:101:10
    |
101 |     ) -> Poll<Option<Self::Item>> {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:120:10
    |
120 |     ) -> Poll<Option<Self::Item>> {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:191:12
    |
191 |         -> Poll<Option<Result<Self::Ok, Self::Error>>>
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:208:14
    |
208 |         ) -> Poll<Option<Self::Item>> {
    |              ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/stream.rs:224:14
    |
224 |         ) -> Poll<Option<S::Item>> {
    |              ^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/mod.rs:9:31
  |
9 | pub use core::task::{Context, Poll, Waker, RawWaker, RawWakerVTable};
  |                               ^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/mod.rs:9:37
  |
9 | pub use core::task::{Context, Poll, Waker, RawWaker, RawWakerVTable};
  |                                     ^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
 --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/__internal/atomic_waker.rs:5:5
  |
5 | use core::task::Waker;
  |     ^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/__internal/atomic_waker.rs:91:30
   |
91 |     waker: UnsafeCell<Option<Waker>>,
   |                              ^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/__internal/atomic_waker.rs:203:29
    |
203 |         impl AssertSync for Waker {}
    |                             ^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/__internal/atomic_waker.rs:261:36
    |
261 |     pub fn register(&self, waker: &Waker) {
    |                                    ^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
   --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/task/__internal/atomic_waker.rs:367:34
    |
367 |     pub fn take(&self) -> Option<Waker> {
    |                                  ^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:12:62
   |
12 | pub type BoxFuture<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + Send + 'a>>;
   |                                                              ^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:16:67
   |
16 | pub type LocalBoxFuture<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + 'a>>;
   |                                                                   ^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:76:30
   |
76 |     where F: ?Sized + Future<Output = Result<T, E>>
   |                              ^^^^^^^^^^^^^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:82:69
   |
82 |     fn try_poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<F::Output> {
   |                                                                     ^^^^^^^^^

error[E0658]: use of unstable library feature 'futures_api': futures in libcore are unstable (see issue #50547)
  --> /Users/rileytg/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.1/src/future.rs:52:57
   |
52 |     impl<F, T, E> Sealed for F where F: ?Sized + Future<Output = Result<T, E>> {}
   |                                                         ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 36 previous errors

Some errors occurred: E0432, E0658.
For more information about an error, try `rustc --explain E0432`.
error: Could not compile `futures-core`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `bandwhich v0.6.0`, intermediate artifacts can be found at `/var/folders/37/jbt828fj49932z8nspl09nc00000gn/T/cargo-installz0gwr1`

Caused by:
  build failed
```

--- a/README.md
+++ b/README.md
@@ -25,6 +25,10 @@ nix-env -iA nixpkgs.bandwhich
 
 #### MacOS and other Linux flavours
 
+`bandwhich` can be installed using the Rust package manager, cargo. If it's not in your distro repositories or the available version is too old, you can install it via [rustup](https://rustup.rs/). You can find additional installation instructions [here](https://doc.rust-lang.org/book/ch01-01-installation.html).
+
+The minimum supported Rust version is **1.39.0**.
+
 ```
 cargo install bandwhich
 ```

